pilot/pkg/xds/eds.go

```diff
@@ -315,11 +315,9 @@ func (s *DiscoveryServer) generateEndpoints(b EndpointBuilder) *endpoint.Cluster
 		return buildEmptyClusterLoadAssignment(b.clusterName)
 	}
 
-	// If networks are set (by default they aren't) apply the Split Horizon
-	// EDS filter on the endpoints
-	if b.push.NetworkManager().IsMultiNetworkEnabled() {
```

Comment:
nit: I prefer this way

Reply:
I'd rather have this sort of logic bundled into the network filter and always apply it. It makes the call site simpler and pushes all the networking-related decisions down into the filter.